table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/CQLStashTableDAO.java

```diff
@@ -0,0 +1,307 @@
+        // Since the range falls entirely within a single shard run a targeted query that only looks for the beginning
+        // of that shard.
+
+        ByteBuffer startToken = RowKeyUtils.getRowKeyRaw(RowKeyUtils.getShardId(fromInclusive), RowKeyUtils.getTableUuid(fromInclusive), new byte[0]);
+
+        ResultSet resultSet = _placementCache.get(_systemTablePlacement)
+                .getKeyspace()
+                .getCqlSession()
+                .execute(
+                        QueryBuilder.select("table_json")
+                                .from(STASH_TOKEN_RANGE_TABLE)
+                                .where(QueryBuilder.eq("stash_id", stashId))
+                                .and(QueryBuilder.eq("data_center", _dataCenters.getSelf().getName()))
+                                .and(QueryBuilder.eq("placement", placement))
+                                .and(QueryBuilder.eq("range_token", startToken))
+                                .and(QueryBuilder.eq("is_start_token", true))
+                                .limit(1)
+                                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM));
+
+        Row row = resultSet.one();
+
+        if (row == null) {
+            return Iterators.emptyIterator();
+        }
+
+        TableJson tableJson = toTableJson(row.getString(0));
+        return Iterators.singletonIterator(new ProtoStashTokenRange(fromInclusive, toExclusive, tableJson));
+    }
+
+    private Iterator<ProtoStashTokenRange> getTokenRangesBetweenInterShard(String stashId, String placement,
+                                                                           ByteBuffer fromInclusive, ByteBuffer toExclusive) {
+        // Since the range crosses the boundary between at least two tables it is guaranteed that if any table has a shard
+        // whose token range intersects the query range then by querying for all start and end tokens within that range
+        // either the start token, end token, or both will be returned in the results.
+
+        ResultSet resultSet = _placementCache.get(_systemTablePlacement)
+                .getKeyspace()
+                .getCqlSession()
+                .execute(
+                        QueryBuilder.select("range_token", "is_start_token", "table_json")
+                                .from(STASH_TOKEN_RANGE_TABLE)
+                                .where(QueryBuilder.eq("stash_id", stashId))
+                                .and(QueryBuilder.eq("data_center", _dataCenters.getSelf().getName()))
+                                .and(QueryBuilder.eq("placement", placement))
+                                .and(QueryBuilder.gte("range_token", fromInclusive))
+                                .and(QueryBuilder.lt("range_token", toExclusive))
+                                .setConsistencyLevel(ConsistencyLevel.LOCAL_QUORUM)
+                                .setFetchSize(100));
```

Comment:
Again, how are we guaranteeing the (From, To, From, To, ...) Order which helps in our Range iterator building?